packages/plugins/Export/src/store/actions.ts

```diff
@@ -12,98 +97,23 @@ const actions: PolarActionTree<ExportState, ExportGetters> = {
       return
     }
 
-    // PDF options
-    const dims = {
-      a0: [1189, 841],
-      a1: [841, 594],
-      a2: [594, 420],
-      a3: [420, 297],
-      a4: [297, 210],
-      a5: [210, 148],
-    }
-
-    //  Screenshot canvas
-    const CANVAS_ID = 'export-canvas'
-
-    map.getInteractions().forEach((interaction) => {
-      interaction.setActive(false)
-    })
+    map.getInteractions().forEach((interaction) => interaction.setActive(false))
 
     map.once('postrender', function () {
-      // Map properties
       const size = map.getSize()
       if (!Array.isArray(size) || size.length !== 2) {
         throw Error('Export: Map has no size.')
       }
-      const viewResolution = map.getView().getResolution()
-      // The canvas to render the screenshot in
-      const mapCanvas: HTMLCanvasElement = document.createElement('canvas')
-      mapCanvas.id = CANVAS_ID
-      // mapCanvas.setAttribute('crossOrigin', 'anonymous')
-      mapCanvas.width = size[0]
-      mapCanvas.height = size[1]
-
-      // Write on this
-      const mapContext = mapCanvas.getContext('2d')
-      if (!mapContext) {
-        console.error(
-          '@polar/plugin-export: map does not of a 2d context, export failed.'
-        )
-        return
-      }
-      // For every ol-layer, get matrix and apply on canvas
-      Array.prototype.forEach.call(
-        document
-          .querySelector('[data-app]')
-          ?.querySelectorAll('.ol-layer canvas'),
-        function (canvas) {
-          if (canvas.width > 0) {
-            const opacity = canvas.parentNode.style.opacity
-            mapContext.globalAlpha = opacity === '' ? 1 : Number(opacity)
-            const transform = canvas.style.transform
-            // Get the transform parameters from the style's transform matrix
-            const matrix = transform
-              .match(/^matrix\(([^(]*)\)$/)[1]
-              .split(',')
-              .map(Number)
-            // Apply the transform to the export map context
-            CanvasRenderingContext2D.prototype.setTransform.apply(
-              mapContext,
-              matrix
-            )
-            mapContext.drawImage(canvas, 0, 0)
-          } else console.warn('@polar/plugin-export: canvas width is 0')
-        }
-      )
-
-      let src = mapCanvas.toDataURL(
-        type === ExportFormat.PNG ? 'image/png' : 'image/jpeg'
-      )
+      let src = getDataUrl(size, type)
+      if (!src) return
 
       if (type === ExportFormat.JPG || type === ExportFormat.PNG) {
-        if (download) {
-          const link = document.createElement('a')
-          link.download = 'map.' + (type === ExportFormat.PNG ? 'png' : 'jpeg')
-          link.href = src
-          document.body.appendChild(link)
-          link.click()
-          document.body.removeChild(link)
-        }
+        if (download) downloadAsImage(src, type)
```

Comment:
As I can' suggest it (deleted lines and such), here is what I would've anticipated:

```ts
if (
  (type === ExportFormat.JPG || type === ExportFormat.PNG) &&
  download
) {
  downloadAsImage(src, type)
```

Reply:
That would be a change in meaning. The else branch would be taken under different circumstances than it is now.

Reply:
It would have a change in meaning, but I highly doubt we'll ever have additional features in this if-clause. You can 🎩 though, if you want.